src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -83,6 +86,24 @@ export default class QuestionnaireForm extends Component {
 
   componentDidMount() {
     this.loadAndMergeForms(this.state.savedResponse);
+
+    const initialFormErrorsNumber = LForms.Util.checkValidity() == null ? 0 : LForms.Util.checkValidity().length;
+    this.setState({
+      formErrorsLength: initialFormErrorsNumber
+    });
+
+    document.addEventListener('change', event => {
+      if(this.props.filterChecked && event.target.id != "filterCheckbox" && event.target.id != "attestationCheckbox") {
+        let newFormErrorNumber = LForms.Util.checkValidity() == null ? 0 : LForms.Util.checkValidity().length;
```

Comment:
Looking into it more, it seems like if an input is invalid, it won't be filtered.  But if you input a valid input somewhere else, it will filter the invalid input and the valid input both at the same time.  The behavior is rather strange, is there a way to check whether the input is valid at the filter function, when it's actually iterating through each question?

Reply:
LForms.Util.checkValidity() returns an array of strings as the errors. It doesn't tell which one is which. I am using the count of the errors to figure out whether to filter or not and it is not very accurate as it includes all the required fields etc. I will look more into the issue.

Reply:
if it's not possible with the LForm api, then we shouldn't worry too much about it.  It's a very minor issue.

Reply:
@KeeyanGhoreshi I have added more logic to check if the field is valid and hopefully it can cover most of the cases. Thanks.